apps/site/lib/site_web/controllers/schedule/finder_api.ex

```diff
@@ -0,0 +1,370 @@
+defmodule SiteWeb.ScheduleController.FinderApi do
+  @moduledoc """
+    API for retrieving journeys for a route, and for
+    showing trip details for each journey on demand.
+
+    "Departure" here is analagous to PredictedSchedule.t()
+  """
+  use SiteWeb, :controller
+
+  alias Predictions.Prediction
+  alias Routes.Route
+  alias Schedules.{Schedule, Trip}
+  alias Site.TransitNearMe
+  alias SiteWeb.ScheduleController.TripInfo, as: Trips
+  alias SiteWeb.ScheduleController.VehicleLocations, as: Vehicles
+
+  import SiteWeb.ScheduleController.ScheduleApi, only: [format_time: 1, fares_for_service: 4]
+
+  @type react_keys :: :date | :direction | :is_current
+  @type react_strings :: [{react_keys, String.t()}]
+  @type converted_values :: {Date.t(), integer, boolean}
+
+  @type enhanced_journey :: %{
+          departure: PredictedSchedule.t() | nil,
+          arrival: PredictedSchedule.t() | nil,
+          trip: Trip.t() | nil,
+          realtime: TransitNearMe.time_data() | nil
+        }
+
+  # How many seconds a departure is considered recent
+  @recent_departure_retention_policy 600
```

Comment:
`retention_policy` sounds like it holds something more complicated than an integer. Perhaps `recent_departure_max_age`?

Reply:
`AbstractRetentionFactoryManagerFactory`